src/zenml/model/artifact_config.py

```diff
@@ -0,0 +1,237 @@
+            artifact_name = artifact.name
+
+        # Create a request model for the model version artifact link
+        request = ModelVersionArtifactRequestModel(
+            user=client.active_user.id,
+            workspace=client.active_workspace.id,
+            name=artifact_name,
+            artifact=artifact_uuid,
+            model=self.model.id,
+            model_version=self.model_version.id,
+            is_model_object=is_model_object,
+            is_deployment=is_deployment,
+            overwrite=self.overwrite,
+        )
+
+        # Create the model version artifact link using the ZenML client
+        existing_links = client.zen_store.list_model_version_artifact_links(
+            ModelVersionArtifactFilterModel(
+                user_id=client.active_user.id,
+                workspace_id=client.active_workspace.id,
+                name=artifact_name,
+                model_id=self.model.id,
+                model_version_id=self.model_version.id,
+                only_artifacts=not (is_model_object or is_deployment),
+                only_deployments=is_deployment,
+                only_model_objects=is_model_object,
+            )
+        )
+        if len(existing_links) and self.overwrite:
+            # delete all model version artifact links by name
+            logger.warning(
+                f"Existing artifact link(s) `{artifact_name}` found and will be deleted."
+            )
+            client.zen_store.delete_model_version_artifact_link(
+                model_name_or_id=self.model.id,
+                model_version_name_or_id=self.model_version.id,
+                model_version_artifact_link_name_or_id=artifact_name,
+            )
+        else:
+            logger.info(
+                f"Artifact link `{artifact_name}` already exists, adding new version."
+            )
+        client.zen_store.create_model_version_artifact_link(request)
+
+    def link_to_model(
+        self,
+        artifact_uuid: UUID,
+    ) -> None:
+        """Link artifact to the model version.
+
+        Args:
+            artifact_uuid (UUID): The UUID of the artifact to link.
+        """
+        self._link_to_model_version(
+            artifact_uuid,
+            is_model_object=False,
+            is_deployment=False,
+        )
+
+
+class ModelArtifactConfig(ArtifactConfig):
+    """Used to link a Model Object to the model version."""
+
+    save_to_model_registry: bool = True
+
+    def link_to_model(
+        self,
+        artifact_uuid: UUID,
+    ) -> None:
+        """Link model object to the model version.
+
+        Args:
+            artifact_uuid (UUID): The UUID of the artifact to link.
+        """
+        self._link_to_model_version(
+            artifact_uuid,
+            is_model_object=True,
+            is_deployment=False,
+        )
+
+
+class DeploymentArtifactConfig(ArtifactConfig):
+    """Used to link a Deployment to the model version."""
```

Comment:
Dumb question: what is a deployment in the context of ZenML? or what is a `DeploymentArtifact` maybe?

Reply:
This is `DeploymentService` in the current status quo, not sure where it will land after the "Complete Model Story" thingy. E.g. `bentoml_model_deployer_step` returns `BentoMLDeploymentService` which can be such an artifact.

Reply:
Maybe this was discussed already, (or maybe coming up with 'complete model story'?) but not sure if a `Service` is an artifact. Give that it's not necessarily something we can version or 'store', somehow feels like it's a separate thing? But you can probably ignore this comment as I wasn't too deep in the discussions you had on this.

Reply:
No, this could be an interesting one 🙂 
What is a Service if not an Artifact - I assumed it is and here how one of them looks in current UI:
<img width="1211" alt="image" src="https://github.com/zenml-io/zenml/assets/31008759/f1787162-8bff-434d-91de-432e7a43e135">

We will have improved UI + enrichment by metadata, but Service still is an Artifact, IMO.

Am I wrong here?

Reply:
Ah ok so we already have such a thing :) Then ignore my comment. I wasn't aware of that. (Though to keep in the back of your head; I'm not sure they're the same kind of thing somehow...)